### PR TITLE
add margin to vertical stepper

### LINKED ISSUE #50635: [STEPPER] Need to custom margin on vertical stepper
Hello Flutter Team,

I realy need this mini evolution to make beautifuler app

## Proposal
Widget : Stepper
Proposal : be able to custom the margin for vertical stepper

In `stepper.dart`, the margin is hard set.
channel beta, line 552
```
            margin: const EdgeInsetsDirectional.only(
              start: 60.0,
              end: 24.0,
              bottom: 24.0,
```
 We need to be able to send our custom margin.
It seems  very simple with an additional parameter for the widget

Thank you flutter

### PR DESCRIPTION
**Pull request by Abhishek01039**
Fixes https://github.com/flutter/flutter/issues/50635

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/stepper.dart [hunk 1]
```diff
@@ -284,6 +285,9 @@ class Stepper extends StatefulWidget {
   /// The elevation of this stepper's [Material] when [type] is [StepperType.horizontal].
   final double? elevation;
 
+  /// Default value for [Stepper.margin].
```
Comment (craiglabenz) [inline packages/flutter/lib/src/material/stepper.dart:288]: I'm guessing this was a placeholder, but I would like to see this docstring describe _where_ the margin will appear.

**Review by craiglabenz — COMMENTED** [review]
This seems like a good start, but there isn't currently any mention of handling the horizontal case. Should this new margin parameter play a role in that Stepper variant as well?

File: packages/flutter/lib/src/material/stepper.dart (hunk 1 above)
Comment (Abhishek01039) [reply packages/flutter/lib/src/material/stepper.dart:288]: Do you have suggestions?

**Comment by Abhishek01039** [comment]
The issue which I am trying to fix is only for vertical I think

File: packages/flutter/lib/src/material/stepper.dart (hunk 1 above)
Comment (craiglabenz) [reply packages/flutter/lib/src/material/stepper.dart:288]: If the target is only the vertical mode, specifying that would be a good start. And then beyond that, a description of where within the Stepper (there are a lot of moving parts!) the extra margin will appear.

**Pull request merged**